core/queryalgebra/model/src/main/java/org/eclipse/rdf4j/query/algebra/BindingSetAssignment.java

```diff
@@ -49,13 +49,34 @@
 	}
 
 	@Override
-	public boolean equals(Object other) {
-		return other instanceof BindingSetAssignment;
+	public boolean equals(Object obj) {
+		if (this == obj)
+			return true;
+		if (obj == null)
+			return false;
+		if (getClass() != obj.getClass())
+			return false;
+		BindingSetAssignment other = (BindingSetAssignment) obj;
+		if (bindingNames == null) {
+			if (other.bindingNames != null)
+				return false;
+		} else if (!bindingNames.equals(other.bindingNames))
+			return false;
+		if (bindingSets == null) {
+			if (other.bindingSets != null)
+				return false;
+		} else if (!bindingSets.equals(other.bindingSets))
+			return false;
+		return true;
 	}
 
 	@Override
 	public int hashCode() {
-		return "BindingSetAssignment".hashCode();
+		final int prime = 31;
+		int result = 1;
+		result = prime * result + ((bindingNames == null) ? 0 : bindingNames.hashCode());
+		result = prime * result + ((bindingSets == null) ? 0 : bindingSets.hashCode());
+		return result;
```

Comment:
This looks like a great catch! Any chance that Objects.hash(....) could be used? https://docs.oracle.com/javase/7/docs/api/java/util/Objects.html#hash(java.lang.Object...)

Reply:
@hmottestad Updated the code. Also replaced Object.equals with Objects.equals in equals() implementation.